src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/QuizManager.java

```diff
@@ -128,23 +130,37 @@ public IsaacQuizDTO findQuiz(final String quizId) throws ContentManagerException
                 throw new ContentManagerException("Expected an IsaacQuizDTO, got a " + contentDTO.getType());
             }
         } else {
-            throw new ContentManagerException("Expected an IsaacQuiz, got a " + cachedContent.getType());
+            if (cachedContent != null) {
+                throw new ContentManagerException("Expected an IsaacQuiz (id=" + quizId + "), got a " + cachedContent.getType());
+            } else {
+                throw new ContentManagerException("Couldn't find quiz with id " + quizId);
+            }
```

Comment:
Have you noticed that the static analysis is saying warning of a useless null check? https://lgtm.com/projects/g/isaacphysics/isaac-api/rev/pr-3f4ff4ea4a09034c27f4a9c4b5e580137c11d2e1
Looks like it might be an odd merge as it is not true on this branch.

Reply:
Yeap, it's from master. I think you noticed the NPE and fixed it on master, and I fixed it separately but in a slightly different way, which is now redundant.

However, I don't like the `null` return, so I've made a combined fix on this branch.

Also, I had named a commit incorrectly so I've rebased this branch.

Reply:
Great, thanks!